src/Cli/dotnet/commands/dotnet-workload/uninstall/WorkloadUninstallCommand.cs

```diff
@@ -50,38 +51,51 @@ internal class WorkloadUninstallCommand : WorkloadCommandBase
 
         public override int Execute()
         {
+            WorkloadHistoryRecorder recorder = new WorkloadHistoryRecorder(_workloadResolver, _workloadInstaller);
+            recorder.HistoryRecord.CommandName = "uninstall";
+            recorder.HistoryRecord.WorkloadArguments = _workloadIds.Select(id => id.ToString()).ToList();
+
             try
             {
-                Reporter.WriteLine();
-
-                var featureBand = new SdkFeatureBand(_sdkVersion);
-                var installedWorkloads = _workloadInstaller.GetWorkloadInstallationRecordRepository().GetInstalledWorkloads(featureBand);
-                var unrecognizedWorkloads = _workloadIds.Where(workloadId => !installedWorkloads.Contains(workloadId));
-                if (unrecognizedWorkloads.Any())
+                recorder.Run(() =>
                 {
-                    throw new Exception(string.Format(LocalizableStrings.WorkloadNotInstalled, string.Join(" ", unrecognizedWorkloads)));
-                }
+                    try
+                    {
+                        Reporter.WriteLine();
 
-                foreach (var workloadId in _workloadIds)
-                {
-                    Reporter.WriteLine(string.Format(LocalizableStrings.RemovingWorkloadInstallationRecord, workloadId));
-                    _workloadInstaller.GetWorkloadInstallationRecordRepository()
-                        .DeleteWorkloadInstallationRecord(workloadId, featureBand);
-                }
+                        var featureBand = new SdkFeatureBand(_sdkVersion);
+                        var installedWorkloads = _workloadInstaller.GetWorkloadInstallationRecordRepository().GetInstalledWorkloads(featureBand);
+                        var unrecognizedWorkloads = _workloadIds.Where(workloadId => !installedWorkloads.Contains(workloadId));
+                        if (unrecognizedWorkloads.Any())
+                        {
+                            throw new Exception(string.Format(LocalizableStrings.WorkloadNotInstalled, string.Join(" ", unrecognizedWorkloads)));
+                        }
+
+                        foreach (var workloadId in _workloadIds)
+                        {
+                            Reporter.WriteLine(string.Format(LocalizableStrings.RemovingWorkloadInstallationRecord, workloadId));
+                            _workloadInstaller.GetWorkloadInstallationRecordRepository()
+                                .DeleteWorkloadInstallationRecord(workloadId, featureBand);
+                        }
 
-                _workloadInstaller.GarbageCollect(workloadSetVersion => _workloadResolverFactory.CreateForWorkloadSet(_dotnetPath, _sdkVersion.ToString(), _userProfileDir, workloadSetVersion));
+                        WorkloadInstallCommand.TryRunGarbageCollection(_workloadInstaller, Reporter, Verbosity, workloadSetVersion => _workloadResolverFactory.CreateForWorkloadSet(_dotnetPath, _sdkVersion.ToString(), _userProfileDir, workloadSetVersion));
```

Comment:
What's the reason for this change?  It looks like the `TryRunGarbageCollection` method will catch any exception thrown, so it won't escape to this method, so if there is a failure the workload history record won't record that correctly.

Reply:
I don't actually 100% remember why I did this. I think it was either matching something I'd done elsewhere or confusion from a merge conflict. Regardless, shouldn't be important, so I switched back to calling _workloadInstaller.GarbageCollect directly.